src/agents/agents.py

```diff
@@ -25,8 +25,10 @@ def get_prompt(user: Optional[dict] = None) -> str:
     """Generate the prompt with current date. (Ignore user information)"""
     return f"""You are a Global Nature Watch's Geospatial Agent with access to tools and user provided selections. Think step-by-step to help answer user queries.
 
-CRITICAL INSTRUCTION: You ALWAYS need AOI + dataset + date range to perform analysis. If ANY are missing or unclear, ask for clarification.
-CRITICAL INSTRUCTION: You MUST call tools sequentially, never in parallel. No parallel tool calling allowed.
+CRITICAL INSTRUCTIONS:
+- You ALWAYS need AOI + dataset + date range to perform analysis. If ANY are missing or unclear, ask for clarification.
+- You MUST call tools sequentially, never in parallel. No parallel tool calling allowed.
+- Be proactive in tool calling, do not ask for clarification or user input unless you absolutely need it. For instance, if dates, places, or datasets dont match exactly, warn the user but move forward with the analysis.,
```

Comment:
This goes against the first instruction - so may be reword the first instruction to something around:
- You ALWAYS need AOI + dataset + date range for analysis: if any are missing or unclear, ask for clarification; but if they are present with minor inconsistencies, proceed proactively and include a warning about the assumptions made.

Reply:
@srmsoumya, I updated the prompt. I am changing it to say ask to specify if something is _missing_. Otherwise be proactive.

@01painadam goot idea, but that is not possible at the moment, because dates is an input to dataset picker. So the agent has not seen the yaml before that. We can revise this as a refinement separately.